src/index.jsx

```diff
@@ -36,53 +40,99 @@ export default class Script extends React.Component {
   constructor(props) {
     super(props);
     this.scriptLoaderId = `id${this.constructor.idCount++}`; // eslint-disable-line space-unary-ops, no-plusplus
+    this.pkID = `script-${shortid.generate()}`;
+    this.unmount = () => {
+      const { url, onUnload } = this.props;
+      const observers = this.constructor.scriptObservers[url];
+      const loaded = this.constructor.loadedScripts[url];
+      const errored = this.constructor.erroredScripts[url];
+
+      // If the component is waiting for the script to load, remove the
+      // component from the script's observers before unmounting the component.
+      if (observers) {
+        delete observers[this.scriptLoaderId];
+        // delete (this.constructor.scriptObservers)[url];
+      }
+
+      // If this script has been previously loaded
+      if (loaded) {
+        delete (this.constructor.loadedScripts)[url];
+        const s = document.querySelector(`[data-pkID="${this.pkID}"]`);
+        s.parentNode.removeChild(s);
+        onUnload();
+      }
+
+      // If this script has been previously errored
+      if (errored) {
+        delete (this.constructor.erroredScripts)[url];
+      }
+    };
+    this.generate = () => {
```

Comment:
Please make this a separate function in the component instead of creating it in constructor